app/src/org/commcare/network/HttpRequestGenerator.java

```diff
@@ -183,6 +183,10 @@ public HttpResponse makeKeyFetchRequest(String baseUri, Date lastRequest) throws
             url = url.buildUpon().appendQueryParameter("last_issued", DateUtils.formatTime(lastRequest, DateUtils.FORMAT_ISO8601)).build();
         }
 
+        // include IMEI in key fetch request for auditing large deployments
+        url = url.buildUpon().appendQueryParameter("device_id",

```

Comment:
Looks like getPhoneId() could potentially return null and appendQueryParameter will throw a NPE in that case. Worth building in handling for that event?